### PR TITLE
Tooltip code

### PR DESCRIPTION
**Pull request by jread-usgs**
This is a start on #148 
 - we now bring in d3 into vizlab minified, and can access it as a resource
 - the tooltip code we have been copypasta is now in the package, but updated for d3
 - instead of the tooltip becoming invisible on mouseover, it is just destroyed and rebuilt when needed
 - the tooltip components (tip, text, box) no longer need to be created in a `visualize` step, they are instead created in this function, assuming `#tooltip-group` exists (errors if not)
 
I haven't added any of the improvements Mary made to the code in #168 or the improvements the interns made this summer as noted in #148. But this is now functional and included in the package. To use it, update your viz.yaml:
image

### CODE REVIEW

File: inst/js/tooltip.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      clearTimeout(hoverTimer); //stop when off area
+    }
+  } else {
+    var tooltip_bg = tipG.append('rect').attr("id", "tooltip-box").attr("height", 24).attr("class", "tooltip-box");
+    var tool_pt = tipG.append('path').attr("id", "tooltip-point").attr("d", "M-6,-11 l6,10 l6,-11").attr("class","tooltip-box");
+    var tooltip = tipG.append('text').attr("id", "tooltip-text").attr("dy","-1.1em").attr('text-anchor',"middle").attr("class","tooltip-text-label svg-text").text(text);
+    pt = cursorPoint(evt);
+    pt.x = Math.round(pt.x);
+    pt.y = Math.round(pt.y);
+    var svgWidth = Number(svg.getAttribute("viewBox").split(" ")[2]);
```
Comment (aappling-usgs) [inline inst/js/tooltip.js]: does this line require that `svg` is in the global environment? and if so, is the existence of `svg` guaranteed by vizlab? what happens if a viz has multiple svg figures?

File: inst/js/tooltip.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+var hoverTimer = null;
+var hoverDelay = 400; //ms
+  
+function hovertext(text, evt){
+  var tipG = d3.selectAll('#tooltip-group');
```
Comment (aappling-usgs) [inline inst/js/tooltip.js]: this makes sense, but just curious - is there a jquery alternative? seems like there are tons of selectors: https://api.jquery.com/category/selectors/

Comment (jread-usgs) [reply inst/js/tooltip.js]: yes, we could make this selection w/ jquery too

File: inst/js/tooltip.js (hunk 1 above)
Comment (jread-usgs) [reply inst/js/tooltip.js]: Good question - that happens in animate.js right now, but I don't know how we should handle multiple svgs within one page, which seems likely in the future...

Comment (aappling-usgs) [reply inst/js/tooltip.js]: is it an option to pass in the svg as an argument to `function hovertext(text, evt){`? unless it's super easy, i think we can just create another issue for this and keep chugging, but agreed that it'd be nice to address at some point.

Comment (aappling-usgs) [reply inst/js/tooltip.js]: created an issue at https://github.com/USGS-VIZLAB/vizlab/issues/273

**Pull request merged**